tests/models/detr/test_modeling_detr.py

```diff
@@ -657,27 +702,65 @@ def test_inference_panoptic_segmentation_head(self):
         # verify outputs
         expected_shape_logits = torch.Size((1, model.config.num_queries, model.config.num_labels + 1))
         self.assertEqual(outputs.logits.shape, expected_shape_logits)
-        expected_slice_logits = torch.tensor(
-            [
-                [-18.1523, -1.7592, -13.5019],
-                [-16.8866, -1.4139, -14.1025],
-                [-17.5735, -2.5090, -11.8666],
-            ]
-        ).to(torch_device)
+        expected_slices = Expectations(
+            {
+                ("xpu", 3): torch.tensor(
+                    [
+                        [-18.1523, -1.7592, -13.5019],
+                        [-16.8866, -1.4139, -14.1025],
+                        [-17.5735, -2.5090, -11.8666],
+                    ], device=torch_device),
+                ("cuda", 7): torch.tensor(
+                    [
+                        [-18.1523, -1.7592, -13.5019],
+                        [-16.8866, -1.4139, -14.1025],
+                        [-17.5735, -2.5090, -11.8666],
+                    ], device=torch_device),
+                ("cuda", 8): torch.tensor(
+                    [
+                        [-18.1523, -1.7592, -13.5019],
+                        [-16.8866, -1.4139, -14.1025],
+                        [-17.5735, -2.5090, -11.8666],
+                    ], device=torch_device),
+                ("rocm", (9, 5)): torch.tensor(
+                    [
+                        [-18.1565,  -1.7568, -13.5029],
+                        [-16.8888,  -1.4138, -14.1028],
+                        [-17.5709,  -2.5080, -11.8654]
+                    ], device=torch_device),
+            }
+        )  # fmt: skip
+        expected_slice_logits = expected_slices.get_expectation()
```

Comment:
Same as 1. 2. and please remove debug print :)
Not commenting on the rest of the PR but it applies for all Expectations dict / print ! Thanks